### PR TITLE
fix: use correct parsers for content types

### PR DESCRIPTION
**Pull request by EsadCetiner**
Closes a possible bypass by enabling the appropriate parser for content types. Some rules don't have an parser for the allowed content types since it's not parseable for ModSecurity

### CODE REVIEW

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 1]
```diff
@@ -122,12 +122,12 @@ SecRule REQUEST_FILENAME "@contains /remote.php/dav/files/" \
     ctl:ruleRemoveTargetById=944120;REQUEST_BODY,\
     ctl:ruleRemoveTargetById=944130;REQUEST_BODY,\
     ctl:ruleRemoveTargetById=944240;REQUEST_BODY,\
-    setvar:'tx.allowed_request_content_type=%{tx.allowed_request_content_type} |text/plain| |text/vcard|'"
+    setvar:'tx.allowed_request_content_type=%{tx.allowed_request_content_type} |text/vcard|'"
 
-# File manager: versioning
+# File manager: Restoring an file to an previous/newer version
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:127]:
```suggestion
# File manager: Restoring a file to a previous / newer version
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 2]
```diff
@@ -198,8 +201,7 @@ SecRule REQUEST_METHOD "@streq REPORT" \
         "t:none,\
         ctl:ruleRemoveById=920340"
 
-# FP while trying to open text files in file manager
-# Adding/editing comments to files
+# Posting an comment on an file
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:204]:
```suggestion
# Posting a comment on a file
```

File: plugins/nextcloud-rule-exclusions-before.conf [hunk 3]
```diff
@@ -386,6 +396,39 @@ SecRule REQUEST_FILENAME "@rx /ocs/v[0-9.]+\.php/apps/files_sharing/api/v[0-9]+/
...
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.2.0',\
+    chain"
+    SecRule REQUEST_HEADERS:Content-Type "@beginsWith text/plain" \
+        "t:none,\
+        ctl:requestBodyProcessor=XML,\
+        setvar:'tx.allowed_request_content_type=%{tx.allowed_request_content_type} |text/plain|'"
+
+# Viewing/Editing comments on an file
```
Comment (theseion) [inline plugins/nextcloud-rule-exclusions-before.conf:414]:
```suggestion
# Viewing/Editing comments on a file
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508182.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+---
+meta:
+  author: "Esad Cetiner"
+  description: "Nextcloud Rule Exclusions Plugin"
+  enabled: true
+  name: 9508182.yaml
+tests:
+  - test_title: 9508182-1
+    desc: Viewing comments on an file
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508182.yaml:9]:
```suggestion
    desc: Viewing comments on a file
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508123.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+---
+meta:
+  author: "Esad Cetiner"
+  description: "Nextcloud Rule Exclusions Plugin"
+  enabled: true
+  name: 9508123.yaml
+tests:
+  - test_title: 9508123-1
+    desc: Posting a comment on an file
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508123.yaml:9]:
```suggestion
    desc: Posting a comment on a file
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508182.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                  xmlns:ocs="http://open-collaboration-services.org/ns">
+                  <oc:limit>20</oc:limit>
+                  <oc:offset>0</oc:offset>
+                </oc:filter-comments>
+          version: HTTP/1.1
+          output:
+            no_log_contains: |-
+              id "911100"|id "920420"|id "941160"
+  - test_title: 9508182-2
+    desc: Editing comments on an file
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508182.yaml:37]:
```suggestion
    desc: Editing comments on a file
```

File: tests/regression/nextcloud-rule-exclusions-plugin/9508551.yaml [hunk 7]
```diff
@@ -23,3 +23,37 @@ tests:
             version: HTTP/1.1
           output:
             no_log_contains: id "931130"
+  - test_title: 9508552-2
```
Comment (theseion) [inline tests/regression/nextcloud-rule-exclusions-plugin/9508551.yaml:26]: Aren't these two tests identical?

**Review by theseion — CHANGES_REQUESTED** [review]

File: tests/regression/nextcloud-rule-exclusions-plugin/9508551.yaml (hunk 7 above)
Comment (EsadCetiner) [reply tests/regression/nextcloud-rule-exclusions-plugin/9508551.yaml:26]: Looks like the duplicate tests were already there beforehand, it's fixed now.

**Review by theseion — APPROVED** [review]

**Pull request merged**